modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +843,328 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            v_float32 a1 = v_load_f32(in + x);
+            v_float32 a2 = v_load_f32(in + x + nlanes / 4);
+            v_float32 a3 = v_load_f32(in + x + nlanes / 2);
+            v_float32 a4 = v_load_f32(in + x + 3 * nlanes / 4);
+
+            vx_store(&out[x], v_pack_u(v_pack(v_round(v_absdiff(a1, s)),
+                                              v_round(v_absdiff(a2, s))),
+                                       v_pack(v_round(v_absdiff(a3, s)),
+                                              v_round(v_absdiff(a4, s)))));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c1(const T in[], const float scalar[], T out[],
+                                      const int length)
+{
+    v_float32 s = vx_setall_f32(*scalar);
+    return absdiffc_simd_c1c2c4(in, out, s, length);
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c2(const T in[], const float scalar[], T out[],
+                                      const int width)
+{
+    constexpr int chan = 2;
+    constexpr int size = static_cast<int>(v_float32::nlanes) + 2;
+    int length = width * chan;
+
+    float init[size];
+    for (int i = 0; i < size; ++i)
+    {
+        init[i] = *(scalar + i % chan);
+    }
+    v_float32 s = vx_load(init);
+
+    return absdiffc_simd_c1c2c4(in, out, s, length);
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c3(short* out_ptr, const v_int32& c1,
+                                              const v_int32& c2, const v_int32& c3,
+                                              const v_int32& c4, const v_int32& c5,
+                                              const v_int32& c6)
+{
+    constexpr int nlanes = static_cast<int>(v_int16::nlanes);
+    vx_store(out_ptr, v_pack(c1, c2));
+    vx_store(out_ptr + nlanes, v_pack(c3, c4));
+    vx_store(out_ptr + 2*nlanes, v_pack(c5, c6));
+}
+
+CV_ALWAYS_INLINE void absdiffc_short_store_c3(ushort* out_ptr, const v_int32& c1,
+                                              const v_int32& c2, const v_int32& c3,
+                                              const v_int32& c4, const v_int32& c5,
+                                              const v_int32& c6)
+{
+    constexpr int nlanes = static_cast<int>(v_uint16::nlanes);
+    vx_store(out_ptr, v_pack_u(c1, c2));
+    vx_store(out_ptr + nlanes, v_pack_u(c3, c4));
+    vx_store(out_ptr + 2*nlanes, v_pack_u(c5, c6));
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c3_impl(const T in[], T out[],
+                                           const v_float32& s1, const v_float32& s2,
+                                           const v_float32& s3, const int length)
+{
+    CV_StaticAssert((std::is_same<T, ushort>::value) ||
```

Comment:
Is there `CV_StaticAssert()` support in standalone mode? IE?

Reply:
Changed to `static_assert()`